### PR TITLE
build(trtexec_vendor): fix cmake to find trtexec bin

### PR DESCRIPTION
**Pull request by felixf4xu**
## Description

This pr fixes https://github.com/autowarefoundation/autoware.universe/issues/5453

## Tests performed

cmake build of trtexec_vendor

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by wep21 — APPROVED** [review]
LGTM

**Review by xmfcx — APPROVED** [review]

File: common/trtexec_vendor/CMakeLists.txt [hunk 1]
```diff
@@ -28,7 +28,8 @@ if(TENSORRT_VERSION VERSION_LESS 8.2.1)
 endif()
 
 set(TRTEXEC_DEFAULT_BIN /usr/src/tensorrt/bin/trtexec)
-if(NOT EXISTS TRTEXEC_DEFAULT_BIN)
+if(NOT EXISTS ${TRTEXEC_DEFAULT_BIN})
+  message(STATUS "${TRTEXEC_DEFAULT_BIN} not found, download and build from source...")
   include(FetchContent)
   fetchcontent_declare(tensorrt
     GIT_REPOSITORY https://github.com/NVIDIA/TensorRT
```
Comment (xmfcx) [inline common/trtexec_vendor/CMakeLists.txt:35]: We get this error:
- https://github.com/autowarefoundation/autoware.universe/actions/runs/6884112074/job/18941486981?pr=5464#step:6:309

Looking at the docs: https://cmake.org/cmake/help/latest/module/FetchContent.html

```suggestion
    GIT_REPOSITORY https://github.com/NVIDIA/TensorRT.git
```

We might need to modify this this way.

Comment (wep21) [reply common/trtexec_vendor/CMakeLists.txt]: We still get the same error after adding ".git". I guess there is another issue to solve.

Comment (felixf4xu) [reply common/trtexec_vendor/CMakeLists.txt]: I re-run the cmake locally, it builds ok (cmake fetch and then build).

**Comment by wep21** [comment]
Internet access is disable during build process due to [this PR](https://github.com/autowarefoundation/autoware-github-actions/pull/241).

**Comment by wep21** [comment]
The package was removed by [this PR](https://github.com/autowarefoundation/autoware.universe/pull/5749).

**Pull request closed without merging**